lib/modules/manager/poetry/extract.ts

```diff
@@ -14,13 +14,27 @@ import type { PackageDependency, PackageFile } from '../types';
 import { extractLockFileEntries } from './locked-version';
 import type { PoetryFile, PoetrySection } from './types';
 
+class PoetryGroup {
+  constructor(public name: string) {}
+}
+
 function extractFromSection(
   parsedFile: PoetryFile,
-  section: keyof Omit<PoetrySection, 'source'>,
+  section: keyof Omit<PoetrySection, 'source' | 'group'> | PoetryGroup,
   poetryLockfile: Record<string, string>
 ): PackageDependency[] {
   const deps: PackageDependency[] = [];
-  const sectionContent = parsedFile.tool?.poetry?.[section];
+  let sectionContent;
+  let depType;
+
+  if (section instanceof PoetryGroup) {
+    sectionContent = parsedFile.tool?.poetry?.group[section.name].dependencies;
```

Comment:
```suggestion
    sectionContent = parsedFile.tool?.poetry?.group[section.name]?.dependencies;
```

Is `section.name` guaranteed to be set? Otherwise we need to treat that defensively too

Reply:
Since we explicitly loop over the keys of `group` when calling `extractFromSection` method, I don't believe we can end up with groups with no `name` attribute, since it would mean having no key, hence no associated structure to parse for the group.

If we write the TOML structure slightly differently, this is a bit more obvious:
```toml
[tool.poetry.group]
dev = {dependencies = {dev_dep1 = "^3.0", dev_dep2 = "^3.1"}}
typing = {dependencies = {typing_dep1 = "^3.0", typing_dep2 = "^3.1"}}
```

It's really similar to what we have for dependencies defined in `[tool.poetry.dependencies]`, for instance.

Reply:
Let's make this defensive change